Modal.js

```diff
@@ -0,0 +1,155 @@
+                </section>
+                <footer class="modal-card-foot is-justify-content-end">
+                    <div class="buttons">
+                        <button class="button is-danger" id="modal-primary-button"></button>
+                        <button class="button is-info" id="modal-secondary-button"></button>
+                    </div>
+                </footer>
+            </div>
+        </div> 
+    `
+
+    input = `<input id="modal-input" class="input">`
+
+    constructor(message, title, primaryBtnText, secondaryBtnText) {
+
+        this.title = title ?? 'Attention' 
+        this.message = message ?? 'This is the default alert message. Have a great day!'
+        this.requireUserInput = false
+        this.primaryBtnText = primaryBtnText ?? 'OK'
+        this.secondaryBtnText = secondaryBtnText ?? 'Cancel'
+    }
+
+    #createTemplate() {
+        
+        const existingModal = document.getElementById(`${this.#id}`)
+        existingModal && this.#close()
+
+        const parser = new DOMParser()
+        const parsedContent = parser.parseFromString(this.modal, 'text/html')
+                
+        const modalTitle = parsedContent.querySelector('.modal-card-title')
+        modalTitle.innerText = this.title
+
+        const modalMessage = parsedContent.querySelector('.label')
+        modalMessage.innerText = this.message
+
+        if (this.requireUserInput) {
+
+            const parsedInput = parser.parseFromString(this.input, 'text/html')
+            const input = parsedInput.querySelector('input')
+            const parsedSection = parsedContent.getElementById('modal-card-field')
+            parsedSection.append(input)
+
+        }
+
+        const modalPrimaryButton = parsedContent.getElementById('modal-primary-button')
+        modalPrimaryButton.innerText = this.primaryBtnText
+
+        const modalSecondaryButton = parsedContent.getElementById('modal-secondary-button')
+        modalSecondaryButton.innerText = this.secondaryBtnText
+
+        const modalDiv = parsedContent.querySelector('.modal')
+        document.body.appendChild(modalDiv)
+
+        this.#createCloseEvents() // Listens for events to close the modal
+
+    }
+
+    prompt(funcToRunOnOkay, funcToRunOnCancel) {
+
+        this.requireUserInput = true
+        this.#createTemplate()
+
+        const primaryButton = document.getElementById('modal-primary-button')
+        const secondaryButton = document.getElementById('modal-secondary-button')
+        const userInput = document.getElementById('modal-input')
+
+        if (funcToRunOnOkay) {
+        
+            primaryButton.addEventListener('click', () => {
+
+                funcToRunOnOkay(userInput.value)
+            
+            })
+
+        }
+
+        secondaryButton.addEventListener('click', () => {
+
+            if (funcToRunOnCancel) {
+
+                funcToRunOnCancel(userInput.value)
+                
+            } else { this.#close() }
+
+        })
```

Comment:
I think that the cancel button should always close the modal _and_ run the function if it exists. What do you think?

Reply:
Good point, making the change now.